ocaml/xenopsd/lib/xenops_utils.ml

```diff
@@ -601,7 +601,7 @@ let detect_hypervisor () =
         let minor =
           String.trim (Unixext.string_of_file _sys_hypervisor_version_minor)
         in
-        Some (Xen (major, minor))
+        Some (Xen (int_of_string major, int_of_string minor))
```

Comment:
This will raise an exception if it's wrong.

Using Scanf should work as it ignores the trailing chars in the files:
```ocaml
    | "xen" ->
        let ( let* ) = Option.bind in
        let int_of_file filename =
          let value = Unixext.string_of_file filename in
          Scanf.ksscanf value (fun _ _ -> None) "%d" Option.some
        in
        let* major = int_of_file _sys_hypervisor_version_major in
        let* minor = int_of_file _sys_hypervisor_version_minor in
        Some (Xen (major, minor))

```

Reply:
There is also `int_of_string_opt`, I can use that.

Reply:
Hmm but if I return None here then further up it is going to print a message that you are not running Xen, but that is not true, you are running Xen, we just can't parse its version.
I'll clean this code up a bit more tomorrow, we need to probably unify detect_hypervisor with look_for_xen (its only user) so we can handle and report errors better.

Reply:
Yeah, I'm unconvinced by the capacity to detect non-xen hypervisors in xenopsd